app/move/controllers/create/base.js

```diff
@@ -24,6 +28,39 @@ class CreateBaseController extends FormWizardController {
 
     next()
   }
+
+  async setSummaryData(req, res, next) {
+    try {
+      const currentLocation = req.session.currentLocation
+      const sessionData = req.sessionModel.toJSON()
+      const { person: personDetail, to_location: toLocationId } = sessionData
+      const person = personService.transform(personDetail)
+
+      let toLocationDetail
+
+      if (toLocationId) {
+        toLocationDetail = await referenceDataService.getLocationById(
+          toLocationId
+        )
+      }
+
+      const moveSummary = presenters.moveToMetaListComponent({
+        ...sessionData,
+        from_location: currentLocation,
+        to_location: toLocationDetail,
+      })
+
+      res.locals = {
+        ...res.locals,
```

Comment:
I have a vague memory of reading somewhere that this isn't good. But that might have been `req.session`...

Could the person local also be this:

```js
res.locals.person = personService.transform(person)
```

Saves having to translate the restructured property and feels less to follow.

Reply:
It also feels wrong to need to use the transform method here.

Ideally it would be better to keep this transform function to the services which keeps it closer to the ideal place we wanted it - the API client.

Just looking at the person service and it should probably be called in the `create` and `update` methods there.

I'm happy to extend the change I made to this so that you can just re-use the person object that already exists.

Reply:
Have added [231](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/pull/231) to ensure the person transformation always happens when interacting with the service so should mean you can simplify this a bit more.